application/templates/_helpers.tpl

```diff
@@ -1,28 +1,37 @@
 {{/* vim: set filetype=mustache: */}}
+
 {{/*
 Define the name of the chart/application.
 */}}
 {{- define "application.name" -}}
 {{- default .Chart.Name .Values.applicationName | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
-{{- define "application.labels.selector" -}}
-app: {{ template "application.name" . }}
+{{/*
+Define the name of the chart/application.
+*/}}
+{{- define "application.version" -}}
+{{ regexReplaceAll "[^a-zA-Z0-9_\\.\\-]" .Values.deployment.image.tag "-" | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
-{{- define "application.labels.stakater" -}}
-{{ template "application.labels.selector" . }}
-appVersion: "{{ .Values.deployment.image.tag | trunc 63 | trimSuffix "-" -}}"
-{{- end -}}
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "application.fullname" -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default .Chart.Name .Values.nameOverride }}
```

Comment:
why are we defining it, if its not being used by any template

Reply:
@mustafaStakater plz remove if not required

Reply:
removed